src/appLoader.js

```diff
@@ -17,6 +17,8 @@ window.SATURN_VERSION = process.env.REACT_APP_VERSION
 
 window._ = _
 
-ReactDOM.render(h(Main), appRoot)
-initializeAuth()
-initializeTCell()
+initializeClientId().then(() => {
```

Comment:
The client id needs to be known before the react components are rendered. Let me know if there are concerns / other ideas here.